samples/generated/snippets.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+# Copyright 2024 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to do basic operations using Cloud
+Spanner.
+For more information, see the README.rst under /spanner.
+"""
+
+import time
+
+from google.cloud import spanner
+from google.cloud.spanner_admin_instance_v1.types import spanner_instance_admin
+
+OPERATION_TIMEOUT_SECONDS = 240
+
+
+# [START spanner_create_instance]
+def create_instance(instance_id):
+    """Creates an instance."""
+    spanner_client = spanner.Client()
+
+    config_name = "{}/instanceConfigs/regional-us-central1".format(
+        spanner_client.project_name
+    )
+
+    operation = spanner_client.instance_admin_api.create_instance(
+        parent="projects/{}".format(spanner_client.project),
+        instance_id=instance_id,
+        instance=spanner_instance_admin.Instance(
+            config=config_name,
+            display_name="This is a display name.",
+            node_count=1,
+            labels={
+                "cloud_spanner_samples": "true",
+                "sample_name": "snippets-create_instance-explicit",
+                "created": str(int(time.time())),
+            },
+        ),
+    )
+
+    print("Waiting for operation to complete...")
+    operation.result(OPERATION_TIMEOUT_SECONDS)
+
+    print("Created instance {}".format(instance_id))
+
+
+# [END spanner_create_instance]
+
+
+# [START spanner_create_database_with_default_leader]
+def create_database_with_default_leader(instance_id, database_id, default_leader):
+    """Creates a database with tables with a default leader."""
```

Comment:
Do we want to include the function name in the snippet, where folks have to call the function or do we want to have it as a runnable snippet? 

For example, see the snippet below which can be run directly
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/automl/snippets/batch_predict.py#L21

Instead of 
```
# [START spanner_create_database_with_default_leader]
def create_database_with_default_leader(instance_id, database_id, default_leader):
    """Creates a database with tables with a default leader."""
```
we could have
```
def create_database_with_default_leader(instance_id, database_id, default_leader):
    """Creates a database with tables with a default leader."""
    # [START spanner_create_database_with_default_leader]
    from google.cloud import spanner

    # TODO(developer): Uncomment and set the following variables
    # instance_id = "<replace with useful text for developer>"
    # database_id = "<replace with useful text for developer>"
    # default_leader =  "<replace with useful text for developer>"
    spanner_client = spanner.Client()
```

Reply:
Thanks @parthea, I want to have one change at a time for this work, and follow the syntax above in separate PR for all samples in repository.